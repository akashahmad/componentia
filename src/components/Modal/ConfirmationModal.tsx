import * as React from "react";

import cx from "classnames";

import { colors } from "../../utils/colors";

import { Typography } from "../Typography/Typography";
import { Button } from "../Button/Button";

import { Modal } from "./Modal";

export function ConfirmationModal({
    open = false,
    onClose,
    position = "top",
    title = "",
    desc,
    actionButtonText = "View Details",
    onActionButtonClick,
    icon: Icon,
    iconBgColor = colors.redLight5,
    actionButtonBgColor,
    showOverlay = true,
}: ConfirmationModalProps) {
    const [_open, set_Open] = React.useState(open || false);

    React.useEffect(() => {
        set_Open(open);
    }, [open]);

    const handleCloseClick = () => {
        set_Open(prev => !prev);
        onClose && onClose();
    };

    return (
        <Modal
            open={_open}
            onClose={handleCloseClick}
            showCloseIcon={false}
            position={position}
            className={cx("flex flex-col justify-center items-center")}
            showOverlay={showOverlay}
        >
            {Icon && (
                <div style={{ background: iconBgColor }} className={cx("p-[18px] rounded-full mb-[22px]")}>
                    {<Icon />}
                </div>
            )}
            <div className="text-center">
                <Typography variant="h3">{title}</Typography>
            </div>
            {!Icon && <span className={cx("inline-block w-[90px] h-[3px] bg-primary rounded-xl mt-[18px]")} />}
            {desc && (
                <div className="mt-6 text-center text-primary-text">
                    <Typography variant="body2">{desc}</Typography>
                </div>
            )}
            <div className={cx("w-full flex items-center justify-between mt-[35px]")}>
                <Button
                    className="mr-[9px]"
                    fullWidth
                    variant="outlined"
                    color="tertiary"
                    onClick={handleCloseClick}
                    style={{ borderColor: colors.stroke }}
                >
                    Cancel
                </Button>
                <Button
                    className="ml-[9px]"
                    fullWidth
                    color="primary"
                    onClick={onActionButtonClick}
                    style={{
                        borderColor: colors.stroke,
                        ...(actionButtonBgColor ? { background: actionButtonBgColor } : {}),
                    }}
                >
                    {actionButtonText}
                </Button>
            </div>
        </Modal>
    );
}

interface ConfirmationModalProps {
    open?: boolean;
    onClose?: () => void;
    position?: "top" | "center" | "bottom";
    title: string;
    desc?: string;
    icon?: React.FC;
    actionButtonText?: string;
    actionButtonBgColor?: string;
    iconBgColor?: string;
    onActionButtonClick?: () => void;
    showOverlay?: boolean;
}
