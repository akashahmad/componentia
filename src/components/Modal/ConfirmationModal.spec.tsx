import * as React from "react";

import { render } from "@testing-library/react";
import { ConfirmationModal } from "./ConfirmationModal";

test("ConfirmationModal component is rendering properly", async () => {
    const TEXT = "TEXT";

    const { container } = render(<ConfirmationModal title={TEXT} />);

    expect(container).toMatchSnapshot();
});
