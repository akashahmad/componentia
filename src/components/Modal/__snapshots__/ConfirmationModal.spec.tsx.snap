// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal component is rendering properly 1`] = `
<div>
  <div
    class="absolute translate-x-[-50%] rounded-[20px] p-[50px] shadow-2 bg-white z-50 left-1/2 top-[30px] hidden flex flex-col justify-center items-center"
    style="width: 530px;"
  >
    <h3
      class="text-[24px] font-semibold leading-[30px]"
    >
      TEXT
    </h3>
    <span
      class="inline-block w-[90px] h-[3px] bg-primary rounded-xl mt-[18px]"
    />
    <div
      class="w-full flex items-center justify-between mt-[35px]"
    >
      <button
        class="rounded-md px-7 py-3 border font-medium flex justify-center items-center w-full text-base px-7 py-3 loading-6 text-dark border-dark hover:bg-grey-4 cursor-pointer mr-[9px]"
        role="button"
        style="border-color: #dfe4ea;"
      >
        Cancel
      </button>
      <button
        class="rounded-md px-7 py-3 border font-medium flex justify-center items-center w-full text-base px-7 py-3 loading-6 bg-primary text-white hover:bg-blue-dark cursor-pointer ml-[9px]"
        role="button"
        style="border-color: #dfe4ea;"
      >
        View Details
      </button>
    </div>
  </div>
</div>
`;
